sfge/src/main/java/com/salesforce/graph/build/VertexSynthesisUtil.java

```diff
@@ -0,0 +1,108 @@
+package com.salesforce.graph.build;
+
+import com.salesforce.apex.jorje.ASTConstants.NodeType;
+import com.salesforce.collections.CollectionUtil;
+import com.salesforce.exception.ProgrammingException;
+import com.salesforce.graph.Schema;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeSet;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+
+/**
+ * Utility class for the standardized addition of properties to Synthetic graph vertices.
+ *
+ * <p>TODO: Parts of {@link StaticBlockUtil} can be refactored into this class.
+ *
+ * <p>TODO: This class has redundant code that could stand to be cleaned up.
+ */
+public class VertexSynthesisUtil {
+
+    public static void addBlockStmtProperties(
+            GraphTraversalSource g, String definingType, Vertex blockStmt, boolean isStandard) {
+        verifyType(blockStmt, NodeType.BLOCK_STATEMENT);
+
+        final Map<String, Object> properties = new HashMap<>();
+        if (isStandard) {
+            properties.put(Schema.IS_STANDARD, true);
+        }
+        properties.put(Schema.IS_SYNTHETIC, true);
+        properties.put(Schema.DEFINING_TYPE, definingType);
+        // ASSUMPTION: We're only ever adding synthetic block statements to the tail of their
+        // parent.
+        properties.put(Schema.CHILD_INDEX, 1);
```

Comment:
It's always unclear when index's or any hard coded number appear in the code.  If it makes sense, do consider defining a constant with appropriate name

Reply:
Basically, the situation where we'd be creating a fake `BlockStatement` vertex is where a method has no body. In that case, the method still has a `Modifier` vertex as its 0th child, so the new `BlockStatement` needs to be added as the 1-indexed child.
I'm not sure a constant makes sense here, but I can at least rewrite the comment to be more descriptive.